boot/bootutil/src/bootutil_misc.c

```diff
@@ -262,26 +259,26 @@ boot_find_status(int image_index, const struct flash_area **fap)
      */
 
     for (i = 0; i < sizeof(areas) / sizeof(areas[0]); i++) {
-        rc = flash_area_open(areas[i], fap);
-        if (rc != 0) {
-            return rc;
-        }
+        uint8_t magic[BOOT_MAGIC_SZ];
 
-        off = boot_magic_off(*fap);
-        rc = flash_area_read(*fap, off, magic, BOOT_MAGIC_SZ);
-        flash_area_close(*fap);
+        if (flash_area_open(areas[i], fap)) {
+            break;
+        }
 
-        if (rc != 0) {
-            return rc;
+        if (flash_area_read(*fap, boot_magic_off(*fap), magic, BOOT_MAGIC_SZ)) {
+            flash_area_close(*fap);
+            break;
         }
 
         if (BOOT_MAGIC_GOOD == boot_magic_decode(magic)) {
             return 0;
         }
 
+        flash_area_close(*fap);
```

Comment:
Shouldn't this `flash_area_close` happen before the test for good magic above?

Reply:
No, the area needs to remain open. I have  have introduced bug with this commit https://github.com/mcu-tools/mcuboot/pull/1183/files, as `boot_find_status` should leave the area open.
Should I revert the change first and then continue with the rework?

Reply:
I don't remember #1183, but the change makes sense if `fap` is the function's "return". I suppose it didn't break anything because "close" doesn't really do much! :-P